src/util.rs

```diff
@@ -9,6 +9,19 @@
 //! Common types used in multiple modules.
 
 use binrw::binrw;
+use thiserror::Error;
+
+///  enumerates all possible errors returned by this library.
+#[derive(Error, Debug)]
+pub enum RekordcrateError {
```

Comment:
So adding more variants later is not a breaking change.
```suggestion
#[non_exhaustive]
pub enum RekordcrateError {
```